### PR TITLE
more use of emplace_back in torrent.cpp

### PR DESCRIPTION
**Pull request by aldenml**

### CODE REVIEW

File: src/torrent.cpp [hunk 1]
```diff
@@ -6365,7 +6365,7 @@ namespace libtorrent {
 			// not be included in this list
 			if (peer->associated_torrent().expired()) continue;
 
-			v->push_back(peer_info());
+			v->emplace_back();
```
Comment (thomasyuan) [inline src/torrent.cpp]: v->emplace_back(peer_info());

Comment (aldenml) [reply src/torrent.cpp]: something wrong with `v->emplace_back()`?

Comment (thomasyuan) [reply src/torrent.cpp]: Needn’t put peer_info() as parameter?

Comment (aldenml) [reply src/torrent.cpp]: no, `emplace_back` builds the object in place with whatever parameters are passed, in this case none, then it uses `peer_info()`

**Pull request merged**